dashboard/dashboard.py

```diff
@@ -242,19 +252,44 @@ def analyze_experiment(info, experiments_dir, tmp_data_dir,
 
     status = validate_status(tmp_analysis_dir)
 
+    # collect data to dump to data_*.json
+    dump_data = {}
     # read the analyzed data, append a timestamp field, and copy over to the permanent data dir
     if status['success']:
         data_exists = check_file_exists(tmp_analysis_dir, 'data.json')
         if not data_exists:
             status = {'success': False, 'message': 'No data.json file produced by {}'.format(exp_name)}
         else:
             data = read_json(tmp_analysis_dir, 'data.json')
-            data['timestamp'] = date_str
-            data['tvm_hash'] = tvm_hash
-            write_json(analyzed_data_dir, 'data_{}.json'.format(date_str), data)
-
+            data.update({
+                'timestamp'  : date_str,
+                'tvm_hash'   : tvm_hash,
+            })
+            dump_data.update(data)
+
+    # fetch time spent on the experiment
+    run_status = validate_json(info.exp_status_dir(exp_name), 
+                            'start_time', 
+                            'end_time', 
+                            'time_delta', filename='run.json')
+    # only process valid files
+    keys = run_status.keys()
+    if len(keys) and functools.reduce(lambda x, y: x and y, 
+                                     map(lambda x: x in keys, 
+                                        ('start_time', 'end_time', 'time_delta'))):
+        rs_get = run_status.get
+        dump_data.update({
+            'success'    : rs_get('success'),
+            'start_time' : rs_get('start_time'),
+            'end_time'   : rs_get('end_time'),
+            'time_delta' : rs_get('time_delta')
+        })
+    # dump date iff we have something to write 
+    write_json(analyzed_data_dir, 'data_{}.json'.format(date_str), dump_data) if len(dump_data) else 'NO DATA; PASS'
     info.report_exp_status(exp_name, 'analysis', status)
-    return status['success']
+    # we may transfer the timing info to data.json for a failed experiment, so
+    # here we have finished writing the data, we can set its status to fail.
+    return status['success'] and run_status['success']
```

Comment:
Please pardon me for being unclear before, but I think we can keep the behavior a little simpler: I don't think we should _create_ a nearly empty `data.json` file for a failed run; I think if the run failed, it's okay for only the `run.json` to have the duration. That is, I think we should only transfer the timings to `data.json` if the run succeeded in the first place

Reply:
It could be useful to store the times for failed runs long-term, though. I just don't think creating a `data.json` for them is the best way to do that (I'll give it some thought)